src/arrayMethodSort.js

```diff
@@ -1,11 +1,27 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const ARRAY = [...this];
+    let cmpFunc = compareFunction;
+
+    if (!compareFunction) {
+      cmpFunc = (a, b) => String(a) > String(b);
+    }
+
+    for (let i = 0; i < ARRAY.length - 1; i++) {
+      for (let j = 0; j < ARRAY.length - 1 - i; j++) {
+        if (cmpFunc(ARRAY[j], ARRAY[j + 1]) > 0) {
+          [ARRAY[j], ARRAY[j + 1]] = [ARRAY[j + 1], ARRAY[j]];
+        }
+      }
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      this[i] = ARRAY[i];
+    }
```

Comment:
This loop is replacing the original array's elements with the sorted elements. However, it's not necessary to copy the array back element by element. Instead, you can replace the entire content of `this` with `ARRAY` using `Array.prototype.splice`.